benchmarkapps/Benchmarks/StartupUsingEndpointRouting.cs

```diff
@@ -43,7 +43,7 @@ public void ConfigureServices(IServiceCollection services)
 
         public void Configure(IApplicationBuilder app)
         {
-            app.UseGlobalRouting();
+            app.UseEndpointRouting();
```

Comment:
Did we settle on `UseRouting`?

Reply:
I think we settled at defining  no user-facing API for registering new middleware.

I think a sensible place to land would be to define the middleware as pubternal and register it from MVC.

Reply:
I think we should make sure that we don't leak these `UseXyz` methods into public API in 2.2.0 - could you just do `.UseMiddleware<>` here instead?

Basically we don't want to put the middleware in a public/supported context because then it lives forever. I think a reasonable thing to do might be to pubternal the middleware and then reference them directly from MVC

Reply:
Or you could move these use methods to `.Internal` namespace. I'm open to other ideas, but I want to make sure we don't accidentlly put something in public API by accident.